src/app/core/standard-guidelines-module/standard-guidelines.service.ts

```diff
@@ -2,52 +2,62 @@ import { Injectable } from '@angular/core';
 import { HttpClient, HttpErrorResponse } from '@angular/common/http';
 import * as querystring from 'querystring';
 import { throwError } from 'rxjs';
-import { retry, catchError, map } from 'rxjs/operators';
+import { catchError, map } from 'rxjs/operators';
 import { STANDARD_GUIDELINES_ROUTES } from './standard-guidelines.routes';
 import { FrameworkDocument } from '../../../entity/standard-guidelines/Framework';
 import { SearchItemDocument } from '../../../entity/standard-guidelines/search-index';
 
+interface GuidelinesFilter {
+  text?: string;
+  filterText?: string;
+  year?: string;
+  levels?: string;
+  page?: string;
+  limit?: string;
+  type?: string;
+  frameworks?: string | string[];
+}
+
 @Injectable({
-  providedIn: 'root'
+  providedIn: 'root',
 })
 export class GuidelineService {
+  constructor(public http: HttpClient) {}
 
-  constructor(public http: HttpClient) { }
-
-  getGuidelines(
-    filter?
+  async getGuidelines(
+    filter?: GuidelinesFilter,
   ): Promise<{ total: number; results: SearchItemDocument[] }> {
     let query = '';
     if (filter) {
       query = querystring.stringify(this.formatFilter(filter));
     }
     // CLARK-SERVICE-FIX: SUPPORT SEARCH QUERY
-    return this.http
-      .get<{ total: number, results: SearchItemDocument[] }>(STANDARD_GUIDELINES_ROUTES.SEARCH_GUIDELINES())
-      .pipe(
-
-        catchError(this.handleError)
-      )
-      .toPromise()
-      .then((res: { total: number, results: any[] }) => {
-        return res;
-      });
+    const res = await this.http
+      .get<{
+        total: number;
+        results: SearchItemDocument[];
+      }>(STANDARD_GUIDELINES_ROUTES.SEARCH_GUIDELINES())
+      .pipe(catchError(this.handleError))
+      .toPromise();
+    return res;
   }
 
   // CLARK-SERVICE-FIX: SUPPORT SEARCH QUERY
   async getFrameworks(query?: any): Promise<FrameworkDocument[]> {
```

Comment:
optional: remove `any` here and replace it with the appropriate type. i think it might be the same `GuidelinesFilter` ([see here](https://github.com/Cyber4All/standard-guidelines-service/blob/main/src/modules/search-module/types/SearchQueryArgs.ts#L4)) though i'm not too sure. not too big of a deal if not you can't find the type fort his.

Reply:
+1